### PR TITLE
Feature pf applauncher #145

### PR DESCRIPTION
**Pull request by akoserwal**
Cleaned up a new branch for this pr. https://github.com/patternfly/patternfly-ng/pull/272

### CODE REVIEW

**Comment by akoserwal** [comment]
https://travis-ci.org/akoserwal/patternfly-ng/builds/334674335

File: src/app/navigation/application-launcher.component.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import { NavigationItemConfig } from './navigation-item-config';
+import { read } from 'fs-extra';
+
+@Component({
+    encapsulation: ViewEncapsulation.None,
+    selector: 'pfng-application-launcher',
+    templateUrl: './application-launcher.component.html'
+})
+
+export class ApplicationLauncherComponent  implements OnInit {
```
Comment (dlabrecq) [inline src/app/navigation/application-launcher.component.ts]: The ApplicationLauncherComponent class needs some doc. Each class and public method shows up in the TS documentation.

File: src/app/navigation/application-launcher.component.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+     * Initialize variable
+     */
+    ngOnInit(): void {
+      this.isClicked = false;
+    }
+
+    /**
+     * toggle function for launcher, active when click, return false on isDisabled:true
+     */
+   public toggle() {
```
Comment (dlabrecq) [inline src/app/navigation/application-launcher.component.ts]: Can this be made private with the above isClicked changes?

File: src/app/navigation/application-launcher.component.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      <li class="applauncher-pf-item" role="menuitem" *ngFor="let item of items">
+        <a class="applauncher-pf-link" href="{{item.url}}" target="{{item.target || '_blank'}}"  *ngFor="let badge of item.badges" title="{{badge.tooltip}}">
+          <i class="applauncher-pf-link-icon pficon" class="{{item.iconStyleClass}}" *ngIf="item.iconStyleClass" [ngClass]="{hidden: hiddenIcons}" aria-hidden="true"></i>
+          <span class="applauncher-pf-link-title">{{item.title}}</span>
+        </a>
+      </li>
+    </ul>
+  </div>
+</div>
+
```
Comment (dlabrecq) [inline src/app/navigation/application-launcher.component.html]: Please move the component files to a directory named  src/app/navigation/application-launcher/

File: src/app/navigation/example/application-launcher-example.component.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            <li class="dropdown">
+              <pfng-application-launcher [label]="'Application Launcher'" [isDisabled]="false" [isList]="false" [items]="navigationItems"
+                [hiddenIcons]="false">
+              </pfng-application-launcher>
+            </li>
+          </ul>
+        </nav>
+      </nav>
+    </div>
+  </div>
```
Comment (dlabrecq) [inline src/app/navigation/example/application-launcher-example.component.html]: The app launcher doesn't appear vertically centered like the Angular Patternfly example. Not sure if it's the height of the example (may be taller than angular-patternfly) or the component's CSS isn't vertically centered?

patternfly-ng:
screen shot 2018-01-29 at 10 38 36 am

angular-patternfly:
screen shot 2018-01-29 at 10 38 46 am

File: src/app/navigation/example/application-launcher-example.component.html [hunk 5]
```diff
@@ -0,0 +1,52 @@
+<div class="padding-15">
```
Comment (dlabrecq) [inline src/app/navigation/example/application-launcher-example.component.html]: The Angular Patternfly example has a switch to show with and without the icons. Can we add that to this example?

File: src/app/navigation/navigation.module.ts [hunk 6]
```diff
@@ -16,8 +17,8 @@ export {
  */
 @NgModule({
   imports: [CommonModule, TooltipModule.forRoot()],
-  declarations: [VerticalNavigationComponent],
-  exports: [VerticalNavigationComponent],
+  declarations: [VerticalNavigationComponent, ApplicationLauncherComponent],
+  exports: [VerticalNavigationComponent, ApplicationLauncherComponent],
```
Comment (dlabrecq) [inline src/app/navigation/navigation.module.ts]: just a nit: alphabetical please

File: src/app/navigation/application-launcher.component.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    /**
+     * The navigation items used to build the menu
+     */
+    @Input() items: NavigationItemConfig[];
+
+
+    /**
+     * Internal boolean to toggle launcher, default:false
+     */
+    public isClicked: boolean;
```
Comment (dlabrecq) [inline src/app/navigation/application-launcher.component.ts]: Please rename as opened.

In addition,  please make this "internal boolean" private. If it cannot be private, due to AOT compilation, then please add an accessor method like so:

// Accessors

get opened(): boolean {
  return this._opened;
}

File: src/app/navigation/application-launcher.component.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    /**
+     *  Use a custom label for the launcher, default: Application Launcher
+     */
+    @Input() label: string;
+
+
+    /**
+     * Disable the application launcher button, default: false
+     */
+    @Input() isDisabled: boolean;
```
Comment (dlabrecq) [inline src/app/navigation/application-launcher.component.ts]: Please rename this, simply as 'disabled'.

File: src/app/navigation/application-launcher.component.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+
+    /**
+     * Display items as a list instead of a grid, default: false
+     */
+    @Input() isList: boolean;
```
Comment (dlabrecq) [inline src/app/navigation/application-launcher.component.ts]: Please rename as showAsList

**Review by dlabrecq — CHANGES_REQUESTED** [review]
Looking good. Some minor comments.

Closing PR #272. FYI, please rebase new changes instead of creating a new PR -- previous comments are lost.

File: src/app/navigation/application-launcher.component.ts (hunk 7 above)
Comment (akoserwal) [reply src/app/navigation/application-launcher.component.ts]: @dlabrecq : isClicked to renamed as opened and accessor method also as opened() ? that's duplicate identifiers

File: src/app/navigation/application-launcher.component.ts (hunk 2 above)
Comment (akoserwal) [reply src/app/navigation/application-launcher.component.ts]: @dlabrecq: I am binding to click event ```(click)="toggle()" ``` and keeping that private won't be allowed by the aot?

File: src/app/navigation/application-launcher.component.html (hunk 3 above)
Comment (akoserwal) [reply src/app/navigation/application-launcher.component.html]: @dlabrecq: sure

File: src/app/navigation/application-launcher.component.ts (hunk 2 above)
Comment (dlabrecq) [reply src/app/navigation/application-launcher.component.ts]: I believe if the 'internal boolean' isClicked/open property is made private, we can probably make the toggle method private as well. If the AOT compilation doesn't allow that, I'm fine with it as is.

**Review by dlabrecq — APPROVED** [review]

**Pull request merged**